### PR TITLE
Builder: Add code to generate a DTFS FDT

### PR DESCRIPTION
**Pull request by ArthurHeymans**

### CODE REVIEW

File: builder/src/lib.rs [hunk 1]
```diff
@@ -46,3 +54,208 @@ fn build_image_with_1_raw_bin_fail() {
...
+    SpiFlash,
+    Emmc,
+    Other,
+}
+
+impl MediumType {
+    fn name(&self) -> &'static str {
+        match self {
+            Self::SpiFlash => "spi-flash",
+            Self::Emmc => "emmc",
```
Comment (MDr164) [inline builder/src/lib.rs]: Most non-x86 boards often allow booting from mmc in general and not just emmc (e.g. SD cards). Should this rather be simplified to mmc, given that SD cards behave rather similar in that regard? Another medium becoming more popular is UFS but we can keep that under 'other' for now.

File: builder/src/lib.rs [hunk 2]
```diff
@@ -46,3 +54,208 @@ fn build_image_with_1_raw_bin_fail() {
...
+    memory_mapping: Option<Vec<MemoryMap>>,
+    medium_type: MediumType,
+    medium_size: u32,
+}
+
+enum HashAlgo {
+    //    Sha256,
+    //    Sha384,
+    Sha512,
+    // TODO are these good targets??
```
Comment (MDr164) [inline builder/src/lib.rs:131]: I've not seen those in firmware before to be honest. But if they are easy to implement I currently don't see a reason to not include those.

File: builder/src/lib.rs [hunk 3]
```diff
@@ -46,3 +54,208 @@ fn build_image_with_1_raw_bin_fail() {
...
+        }
+    }
+}
+
+struct DtfsDigest {
+    algo: HashAlgo,
+    digest: Vec<u8>,
+}
+
+enum CompressionAlgo {
```
Comment (MDr164) [inline builder/src/lib.rs:151]: For u-boot another popular algorithm is DEFLATE but that's arguably worse than zstd and lz4 in my opinion so for now I'm fine with having 'just' those three

File: builder/src/lib.rs (hunk 2 above)
Comment (ArthurHeymans) [reply builder/src/lib.rs:131]: It's the dilithium based fips204 ones. It's to keep the buzzword count high.

File: builder/src/lib.rs (hunk 3 above)
Comment (ArthurHeymans) [reply builder/src/lib.rs:151]: Adding new ones ought to be trivial in rust (where it took me a half a day in C for coreboot)

**Review by MDr164 — APPROVED** [review]
lgtm so far but just noticed one thing: Now that you added the license fields into the Cargo.toml we should also add the actual license file into this repository.

**Comment by ArthurHeymans** [comment]
> lgtm so far but just noticed one thing: Now that you added the license fields into the Cargo.toml we should also add the actual license file into this repository.

I'll do that in a separate PR

**Pull request merged**